### PR TITLE
feat: labeled metrics

### PR DESCRIPTION
**Pull request by swelf19**
i've added a new methods to ValidatorsRepository, Monitor and Collector interfaces, to provide a labeled gauges for prometheus.

GetMetricVectors() map[Metric]map[string]float64 
added for Monitor

DisplayName(string) string
added for ValidatorsRepository

and 
GetVector(metric monitors.Metric) (map[string]float64, error)
ProvidedMetricVectors() []monitors.Metric
added for collector.

as a result we have
Screenshot_20210804_131701

on the screenshot you can see contracts public keys as labels. The DisplayName on validatorsRepository is responsible for the labels.

### CODE REVIEW

**Comment by oopcode** [comment]
image
Looks good, but the list of validators is getting out of its panel box; can we do something about that?

File: collector/monitors/slashing_monitor.go [hunk 1]
```diff
@@ -203,3 +213,9 @@ func (r *V1ValidatorsRepository) GetValidatorsAddresses(ctx context.Context) ([]
 
 	return hubResp.Validators, nil
 }
+
+
+func (r *V1ValidatorsRepository) DisplayName(pubkey string) string {
+	// TODO: implement the real method to match pubkey with display name
+	return pubkey
+}
```
Comment (oopcode) [inline collector/monitors/slashing_monitor.go:221]: missing newline

File: collector/monitors/monitors.go [hunk 2]
```diff
@@ -9,8 +9,10 @@ type Monitor interface {
 	InitMetrics()
 	// Handler fetches the data to inner storage
 	Handler(ctx context.Context) error
-	// GetMetrics - provides metrics fetched by Handler method
+	// GetMetrics - provides single value metrics fetched by Handler method
 	GetMetrics() map[Metric]float64
+	// GetMetricVectors - provides set of a labeled values fetched by Handler method
+	GetMetricVectors() map[Metric]map[string]float64
```
Comment (oopcode) [inline collector/monitors/monitors.go:15]: `map[Metric]map[string]float64` looks a bit cumbersome, maybe we could add a subtype?

File: collector/monitors/slashing_monitor.go (hunk 1 above)
Comment (swelf19) [reply collector/monitors/slashing_monitor.go]: done

File: collector/monitors/monitors.go (hunk 2 above)
Comment (swelf19) [reply collector/monitors/monitors.go]: declared new type
 type MetricVector map[string]float64

**Comment by swelf19** [comment]
> Looks good, but the list of validators is getting out of its panel box; can we do something about that?
not much we can do about.
take validators name as short as possible.
left only validators names in legend panel
Screenshot_20210805_123952

however probably they will not fit vertically

File: collector/monitors/slashing_monitor.go [hunk 3]
```diff
@@ -203,3 +213,10 @@ func (r *V1ValidatorsRepository) GetValidatorsAddresses(ctx context.Context) ([]
 
 	return hubResp.Validators, nil
 }
+
+
+func (r *V1ValidatorsRepository) DisplayName(pubkey string) string {
+	// TODO: implement the real method to match pubkey with display name
+	valName := []rune(pubkey)
```
Comment (pr0n00gler) [inline collector/monitors/slashing_monitor.go:220]: Why not use moniker from validator info endpoint? I think, it's clearer than just a few first characters of a validator address

Comment (swelf19) [reply collector/monitors/slashing_monitor.go]: done
Screenshot_20210805_174733

**Review by oopcode — APPROVED** [review]

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**